plugins/modules/v_switch.py

```diff
@@ -0,0 +1,487 @@
+        module.fail_json(msg=result['error']['message'])
+    elif error == 'VSWITCH_IN_PROCESS':
+        module.fail_json(msg='There is a update running, therefore the vswitch can not be updated')
+
+    wait_condition = is_all_servers_ready if module.params['wait'] else None
+    return get_v_switch(module, id_, wait_condition)
+
+
+def get_servers_to_delete(current_servers, desired_servers):
+    return [
+        server['server_ip']
+        for server in current_servers
+        if server['server_ip'] not in desired_servers
+        and str(server['server_number']) not in desired_servers
+    ]
+
+
+def get_servers_to_add(current_servers, desired_servers):
+    current_ids = [str(server['server_number']) for server in current_servers]
+    current_ips = [server['server_ip'] for server in current_servers]
+    return [
+        server
+        for server in desired_servers
+        if server not in current_ips and server not in current_ids
+    ]
+
+
+def set_desired_servers(module, id_):
+    v_switch = get_v_switch(module, id_)
+    changed = False
+
+    servers_to_delete = get_servers_to_delete(v_switch['server'], module.params['servers'])
+    if len(servers_to_delete) > 0:
+        v_switch = delete_servers(module, id_, servers_to_delete)
+        changed = True
+    servers_to_add = get_servers_to_add(v_switch['server'], module.params['servers'])
+    if len(servers_to_add) > 0:
+        v_switch = add_servers(module, id_, servers_to_add)
+        changed = True
+    return (v_switch, changed)
+
+
+def main():
+    argument_spec = dict(
+        vlan=dict(type='int', required=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        servers=dict(type='list', elements='str', default=[]),
+        wait=dict(type='bool', default=True),
+        wait_delay=dict(type='int', default=10),
+        timeout=dict(type='int', default=180),
+    )
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    v_switches, error = fetch_url_json(module, V_SWITCH_BASE_URL, accept_errors=['UNAUTHORIZED'])
+
+    if error:
+        module.fail_json(msg='Please check your current user and password configuration')
+
+    matched_v_switches = [
+        v
+        for v in v_switches
+        if v['name'] == module.params['name'] and v['vlan'] == module.params['vlan']
+    ]
+    non_cancelled_v_switches = [m for m in matched_v_switches if m['cancelled'] is False]
+    result = {'changed': False}
+
+    if len(non_cancelled_v_switches) > 1:
+        module.fail_json(
+            msg='Multiple vSwitches with same name and VLAN ID in non cancelled status. Clean it'
+        )
+
+    elif len(non_cancelled_v_switches) == 1:
+        id_ = non_cancelled_v_switches[0]['id']
+        v_switch, changed = set_desired_servers(module, id_)
```

Comment:
So in case `state=absent`, you first modify the server list? Is this necessary / does this make sense?

Reply:
vSwitches are cancelled at the end of the day. If you want to remove it and attach other servers in the same VLAN after it you should detach them first from the cancelled vSwitch.

Any way if you don't want to remove them you can keep the list as it was or , currently, you can not define the servers parameter.

Reply:
Shouldn't this then always clear the server list (if `state=absent`), and not only when the user specifies `state: absent` together with `servers: []`?

Reply:
I would like to be consistent with the `servers: []` parameter.

I don't see big differences between both options but this way sounds more consistent for me. I dunno who will be interested in doing it but Hetzner API support cancelling the vSwitch without removing the servers.

Reply:
I guess one reason is that you can still use the vSwitch until it completely disappears :shrug:

How about just documenting this, i.e. mention in the documentation to `state=absent` that you explicitly have to specify `servers=[]` to remove servers from the vSwitch.

Reply:
Added in 776cb459887661c32bd6798df38f1bc551093687